devito/finite_differences/derivative.py

```diff
@@ -504,6 +504,26 @@ def _eval_at(self, func):
             return self
 
         x0 = func.indices_ref.getters
+        # Check if x0's keys come from a DerivedDimension
```

Comment:
Should `x0`'s keys be generated with the root dimension instead? I suppose this approach would allow you to use a subdimension in `x0`